sql/core/src/test/scala/org/apache/spark/sql/test/TestSQLContext.scala

```diff
@@ -34,9 +40,32 @@ private[sql] class TestSparkSession(sc: SparkContext) extends SparkSession(sc) {
     this(new SparkConf)
   }
 
-  @transient
-  override lazy val sessionState: SessionState = new SessionState(self) {
-    override lazy val conf: SQLConf = {
+  class TestSessionState(
+      sparkContext: SparkContext,
+      sharedState: SharedState,
+      conf: SQLConf,
+      experimentalMethods: ExperimentalMethods,
+      functionRegistry: FunctionRegistry,
+      catalog: SessionCatalog,
+      sqlParser: ParserInterface,
+      analyzer: Analyzer,
+      streamingQueryManager: StreamingQueryManager,
+      queryExecution: LogicalPlan => QueryExecution)
+    extends SessionState(
+        sparkContext,
+        sharedState,
+        conf,
+        experimentalMethods,
+        functionRegistry,
+        catalog,
+        sqlParser,
+        analyzer,
+        streamingQueryManager,
+        queryExecution) {}
+
+  object TestSessionState {
+
+    def createTestConf: SQLConf = {
```

Comment:
is this needed any where else? then why make a separate method? just define it inside `apply`